core/src/main/java/org/infinispan/partitionhandling/impl/AvailabilityStrategy.java

```diff
@@ -32,7 +32,10 @@ static ConsistentHash readConsistentHash(CacheTopology topology, ConsistentHashF
          case NO_REBALANCE:
             return topology.getCurrentCH();
          case TRANSITORY:
-            return topology.getPendingCH();
+            // This is used to determine nodes that own the entries. In scattered cache (which uses transitory topology)
+            // the pendingCH is used for reading but the nodes in there are not guaranteed to have the data yet.
+            // CurrentCH should be safe - the nodes either have the data or the owner is unknown.
+            return topology.getCurrentCH();
```

Comment:
I meant this method to be a mirror of `CacheTopology.getReadConsistentHash()`, only without using the `unionCH` field (which is `null` on the `ClusterTopologyManagerImpl` side). Since this is breaking the symmetry, we should change the method name as well.